### PR TITLE
FileUtils.make: don't bypass superenv

### PR DESCRIPTION
**Pull request by mistydemeo**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Updated version of #2495 - this should resolve the problems with that version. Was reminded when working in homebrew-portable again.

cc @xu-cheng - I believe this fixes the issue you noticed with that branch.

### CODE REVIEW

File: Library/Homebrew/shims/super/make [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
 
+export MAKE=${MAKE:-make}
```
Comment (xu-cheng) [inline Library/Homebrew/shims/super/make]: Would it better to use `HOMEBREW_MAKE` as it is a private variable used by Homebrew?

**Comment by xu-cheng** [comment]
:+1:

File: Library/Homebrew/shims/super/make (hunk 1 above)
Comment (mistydemeo) [inline Library/Homebrew/shims/super/make]: This was in response to https://github.com/Homebrew/brew/pull/2495#commitcomment-21896290, which suggested setting this to make sure the right `MAKE` gets called in all circumstances.

**Comment by MikeMcQuaid** [comment]
Thanks again @mistydemeo!

**Pull request merged**